SCClassLibrary/Common/Streams/Patterns.sc

```diff
@@ -93,11 +93,14 @@ Pattern : AbstractFunction {
 	// dur: if nil, record until pattern stops or is stopped externally
 	// fadeTime: allow extra time after last Event for nodes to become silent
 
-	record { |path, headerFormat = "AIFF", sampleFormat = "float", numChannels = 2, dur = nil, fadeTime = 0.2, clock(TempoClock.default), protoEvent(Event.default), server(Server.default), out = 0, outNumChannels|
+	record { |path, headerFormat, sampleFormat, numChannels = 2, dur = nil, fadeTime = 0.2, clock(TempoClock.default), protoEvent(Event.default), server(Server.default), out = 0, outNumChannels|
 
 		var recorder = Recorder(server);
 		var pattern = if(dur.notNil) { Pfindur(dur, this) } { this };
 
+		headerFormat !? { recorder.recHeaderFormat = headerFormat };
```

Comment:
The check for nil isn't necessary, right? When it is nil, it will just use the server's.

So perhaps just: 
```
recorder.recHeaderFormat = headerFormat;
recorder.recSampleFormat = sampleFormat;
```

Reply:
I thought I didn't want to explicitly set those parameters as nil. But actually, on second thought, I don't see any reason why we shouldn't, and we save two nil-checks, so I changed it. Thanks!